velox/functions/prestosql/window/Ntile.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/Exceptions.h"
+#include "velox/exec/WindowFunction.h"
+#include "velox/expression/FunctionSignature.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::window {
+
+namespace {
+
+class NtileFunction : public exec::WindowFunction {
+ public:
+  explicit NtileFunction(
+      const std::vector<TypePtr>& argTypes,
+      const std::vector<column_index_t>& argIndices)
+      : WindowFunction(BIGINT(), nullptr) {
+    if (!argTypes.size() || !argTypes[0]->isBigint() || argIndices[0] <= 0) {
+      VELOX_FAIL("Ntile function requires a positive integer argument");
```

Comment:
Yes, the argIndices[0] > 0 check is there in Presto Java also. I added the other two checks since Presto java expects the argument to be of type BIGINT.